app/governance/proposal/page.tsx

```diff
@@ -0,0 +1,315 @@
+import { NewTransactionFlow } from "@/transactions/flows/types";
+import Spacer from "@/components/layout/spacer";
+
+export default function Page() {
+  function castVote(proposalId: number, voteOption: VoteOption | null) {
+    if (signer) {
+      if (!voteOption) {
+        return NEW_ERROR("Please select a vote option");
+      }
+      //console.log("cast vote test");
+      const newFlow: NewTransactionFlow = {
+        icon: "/canto.svg",
+        txType: TransactionFlowType.VOTE_TX,
+        title: "Voting",
+        params: {
+          chainId: chainId,
+          ethAccount: signer?.account.address ?? "",
+          proposalId: proposalId,
+          voteOption: voteOption,
+        },
+      };
+      txStore?.addNewFlow({
+        txFlow: newFlow,
+        ethAccount: signer.account.address,
+      });
+    }
+  }
+
+  const searchParams = useSearchParams();
+  const id = searchParams.get("id");
+  const proposalId = Number(id);
+  const router = useRouter();
+
+  const { txStore, signer, chainId } = useCantoSigner();
+  const { proposals, isProposalsLoading } = useProposals({ chainId: chainId });
+  const [selectedVote, setSelectedVote] = useState<VoteOption | null>(null);
+
+  const [isChecked, setChecked] = useState(false);
+
+  const handleCheck = (voteOption: VoteOption) => {
+    setSelectedVote(voteOption);
+    setChecked(!isChecked);
+  };
+
+  //console.log(selectedVote);
+  if (isProposalsLoading) {
+    return <Splash />;
+  }
+
+  if (!id) {
+    return (
+      <div>
+        <Text font="proto_mono">Proposal ID is missing</Text>
+      </div>
+    );
+  }
+
+  function getVotingBoxStyles() {
+    if (isActive) {
+      return {
+        height: "70%",
+        padding: "10px 0px 0px 0px",
+      };
+    }
+    return {
+      height: "100%",
+      padding: "0px 0px 20px 0px",
+    };
+  }
+  const proposal = proposals.find((p) => p.proposal_id === Number(proposalId));
+
+  if (!proposal) {
+    return (
+      <div className={styles.noProposalContainer}>
+        <Text font="proto_mono">
+          No proposal found with the ID {proposalId}{" "}
+        </Text>
+      </div>
+    );
+  }
+
+  const isActive = formatProposalStatus(proposal.status) == "ACTIVE";
+
+  const votesData = calculateVotePercentages(proposal.final_vote);
+
+  //console.log(votesData);
+
+  // const colorMap = new Map<number,String>();
+  // colorMap.set(1, "rgba(6, 252, 153, 0.5)");
+  // colorMap.set(2, "rgb(252, 81, 81,0.5)");
+  // colorMap.set(3, "rgb(68, 85, 239,0.5)");
+  // colorMap.set(4, "rgb(111, 105, 105,0.5)");
+  const amounts = [
+    parseFloat(votesData.Yes),
+    parseFloat(votesData.No),
+    parseFloat(votesData.Veto),
+    parseFloat(votesData.Abstain),
+  ];
+  //console.log(Math.max(...amounts));
+
+  const maxAmountIndex = amounts.indexOf(Math.max(...amounts));
+
+  return isProposalsLoading ? (
+    <Splash />
+  ) : (
+    <div className={styles.proposalContainer}>
```

Comment:
@waajito can you look at this? There should be a better way to do styling for this